libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss

```diff
@@ -21,32 +21,62 @@
     }
   }
 
+  &.default-mode {
+    ion-segment {
+      --background: #{utils.get-color('white')};
+
+      display: inline-flex; // The segmented control itself is block-level, but we don't want the background to stretch
+      width: auto; // The segmented control itself is block-level, but we don't want the background to stretch
+      overflow: visible; // Ensures the badge is not cut off
+      contain: unset; // Ensures the badge is not cut off
+    }
+  }
+
   &.chip-mode {
-    display: flex;
-    flex-wrap: nowrap;
-    overflow-x: auto;
-    overflow-y: hidden; // chip touch area height is bigger than segmented control height
-    // Hide scrollbar on touch device:
-    @include utils.touch {
-      scrollbar-width: none; /* Firefox */
-      &::-webkit-scrollbar {
-        display: none;
+    ion-segment {
+      @include utils.touch {
+        scrollbar-width: none; /* Firefox */
+        &::-webkit-scrollbar {
+          display: none;
+        }
+      }
+
+      --background: none;
+
+      border-radius: 0;
+      padding: #{utils.size('xxxs')} #{utils.size('xxxs')};
+      gap: #{utils.size('xxs')};
+    }
+
+    ion-segment-button {
+      --background: #{utils.get-color('white')};
+      --indicator-transition: none;
+
+      flex: initial;
+
+      &::part(indicator) {
+        // Remove padding as the selected segment will otherwise have additional space to its neighbors
+        padding-inline: 0;
       }
     }
   }
 
   &.compact {
-    padding-inline: utils.size('xxxxs');
+    ion-segment {
+      --background: transparent;
+
+      gap: #{utils.size('xxxs')};
+    }
+
+    ion-segment-button {
+      --background: transparent;
+      --padding-start: #{utils.size('xs')};
+      --padding-end: #{utils.size('xs')};
```

Comment:
If you're having issues with compact mode being too compact, you may be able to hit approximately the same amount of padding as before by changing these two custom properties. I think the closest you get is by using `15px` but since that is not one of our size tokens `size('s')` is probably an ok size.

If you choose `size('s')` you could even just remove both declarations, since it's the same as declared in [lines 19-20](https://github.com/kirbydesign/designsystem/blob/7e7c9e4da277169cd466b267501d6be102700f84/libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss#L19-L20). This of course requires that  `.sm` is used with compact mode.

Reply:
Yeah, the reason for the compactness right now is actually that I wanted to propose a slightly _compacter_ mode, as there was very little difference before. But right now, I think it is just too compact.